### DISCUSSION TITLE
Concern about numbers of subject/subscription

### DISCUSSION THREAD

**Discussion by joejoe2** [nats-io/nats-server#4254] [Q&A]
I wonder will nats still has high performance(throughput) of there are large numbers of subject/subscription ? For example, there are millions of users in a chat application and each user has a subject(by id). Each user connection to the application will create a subscription on his subject. Sending messages to another user can be done by publish to his subject. So another user can connect to different application instance and notified new messages across the nats.

**Comment by derekcollison** [comment]
We believe millons of subscriptions should be handled fairly well, we are working with some strategics and customers on being able to directly connect billions of users or things while maintaining location independence.
